reporter/internal/pdata/generate.go

```diff
@@ -0,0 +1,322 @@
+
+// mkProfileID creates a random profile ID.
+func mkProfileID() []byte {
+	profileID := make([]byte, 16)
+	_, err := rand.Read(profileID)
+	if err != nil {
+		return []byte("opentelemetry-ebpf-profiler")
+	}
+	return profileID
+}
+
+// setProfile sets the data an OTLP profile with all collected samples up to
+// this moment.
+func (p *Pdata) setProfile(
+	events map[samples.TraceAndMetaKey]*samples.TraceEvents,
+	profile pprofile.Profile,
+) {
+	// stringMap is a temporary helper that will build the StringTable.
+	// By specification, the first element should be empty.
+	stringMap := make(map[string]int32)
+	stringMap[""] = 0
+
+	// funcMap is a temporary helper that will build the Function array
+	// in profile and make sure information is deduplicated.
+	funcMap := make(map[samples.FuncInfo]int32)
+	funcMap[samples.FuncInfo{Name: "", FileName: ""}] = 0
+
+	// attributeMap is a temporary helper that maps attribute values to
+	// their respective indices.
+	// This is to ensure that AttributeTable does not contain duplicates.
+	attributeMap := make(map[string]int32)
+
+	st := profile.SampleType().AppendEmpty()
+	st.SetTypeStrindex(getStringMapIndex(stringMap, "samples"))
+	st.SetUnitStrindex(getStringMapIndex(stringMap, "count"))
+
+	pt := profile.PeriodType()
+	pt.SetTypeStrindex(getStringMapIndex(stringMap, "cpu"))
+	pt.SetUnitStrindex(getStringMapIndex(stringMap, "nanoseconds"))
+	profile.SetPeriod(1e9 / int64(p.samplesPerSecond))
+
+	// Temporary lookup to reference existing Mappings.
+	fileIDtoMapping := make(map[libpf.FileID]int32)
+
+	var locationIndex int32
+	var startTS, endTS pcommon.Timestamp
+	for traceKey, traceInfo := range events {
+		sample := profile.Sample().AppendEmpty()
+		sample.SetLocationsStartIndex(locationIndex)
+
+		slices.Sort(traceInfo.Timestamps)
+		startTS = pcommon.Timestamp(traceInfo.Timestamps[0])
+		endTS = pcommon.Timestamp(traceInfo.Timestamps[len(traceInfo.Timestamps)-1])
+
+		sample.TimestampsUnixNano().FromRaw(traceInfo.Timestamps)
+		sample.Value().Append(1)
+
+		// Walk every frame of the trace.
+		for i := range traceInfo.FrameTypes {
+			frameAttributes := addProfileAttributes(profile, []samples.AttrKeyValue[string]{
+				{Key: "profile.frame.type", Value: traceInfo.FrameTypes[i].String()},
+			}, attributeMap)
+
+			loc := profile.LocationTable().AppendEmpty()
+			loc.SetAddress(uint64(traceInfo.Linenos[i]))
+			loc.AttributeIndices().FromRaw(frameAttributes)
+
+			switch frameKind := traceInfo.FrameTypes[i]; frameKind {
+			case libpf.NativeFrame:
+				// As native frames are resolved in the backend, we use Mapping to
+				// report these frames.
+
+				var locationMappingIndex int32
+				if tmpMappingIndex, exists := fileIDtoMapping[traceInfo.Files[i]]; exists {
+					locationMappingIndex = tmpMappingIndex
+				} else {
+					idx := int32(len(fileIDtoMapping))
+					fileIDtoMapping[traceInfo.Files[i]] = idx
+					locationMappingIndex = idx
+
+					ei, exists := p.Executables.Get(traceInfo.Files[i])
```

Comment:
Missed fix from https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/247:
```suggestion
					ei, exists := p.Executables.GetAndRefresh(traceInfo.Files[i], ...)
```